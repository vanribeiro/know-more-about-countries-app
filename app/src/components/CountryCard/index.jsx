import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import styled from 'styled-components';
import { Strong, P } from '../UI';
import { lightMode } from '../UI/variables';

const Cardsx = {
    maxWidth: '100%', 
    boxShadow: '1px 1px 5px #e2e2e2'
}

const CardContentsx = {
    padding: '28px 24px'
}

const TitleCard = styled.h2`
    font-size: 16px;
    margin-bottom: 10px;
    color: ${ lightMode.textColor };
`;

const CountryCard = ({info}) => {
    return(
        <>
            <Card sx={Cardsx}>
                <CardMedia 
                    component="img"
                    alt={`${info.name.common} Flag`}
                    height="180"
                    image={`${info.flags.svg}`}
                />
                <CardContent sx={CardContentsx}>
                    <TitleCard>{info.name.common}</TitleCard>
                    <P>
                        <Strong>Population: </Strong> 
                        {(info.population).toLocaleString('en-us')}
                    </P>
                    <P>
                        <Strong>Region: </Strong> 
                        {info.region}
                    </P>
                    <P>
                        <Strong>Capital: </Strong>
                        {
                            info.capital.length === 0 || info.capital === undefined
                            ? "Not Specified" 
                            : info.capital
                        }
                    </P>
                    
                </CardContent>
            </Card>
        </>
    )
}

export default CountryCard;