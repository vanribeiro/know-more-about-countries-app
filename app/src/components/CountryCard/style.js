import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import styled from 'styled-components';

const CardContainer = styled(Card)`
    max-width: 100%;
    box-shadow: 1px 1px 5px ${({theme}) => theme.boxShadowColor};
    background-color: ${({theme}) => theme.cardBackgroundColor};
`;

const CardImage = styled(CardMedia)`

    @media screen and (max-width: 640px){
        height: 180px;
    }

    @media screen and (min-width: 641px){
        height: 160px;
    }

    @media screen and (min-width: 768px){
        height: 120px;
    }

    @media screen and (min-width: 892px){
        height: 150px;
    }

    @media screen and (min-width: 992px){
        height: 144px;
    }

    @media screen and (min-width: 1200px){
        height: 150px;
    }

    @media screen and (min-width: 1400px){
        height: 165px;
    }

`;

const TitleCard = styled.h2`
    font-size: 16px;
    margin-bottom: 10px;
    color: ${({theme}) => theme.text};
`;

const CardContentContainer = styled(CardContent)`
    background-color: ${({theme}) => theme.cardContentBackgroundColor};
    padding: 24px 24px 36px !important;
`;

export {
    TitleCard,
    CardContainer,
    CardContentContainer,
    CardImage
}