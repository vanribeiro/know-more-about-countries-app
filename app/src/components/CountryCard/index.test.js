import React from 'react';
import CountryCard from ".";
import { render } from '@testing-library/react';

describe('Card component', () => {
    it('should always have the same snapshot', () => {
        const info = {
          flags: {
            png: "https://flagcdn.com/w320/me.png",
            svg: "https://flagcdn.com/me.svg"
          },
          name: {
            common: "Montenegro",
          },
          capital: [
            "Podgorica"
          ],
          region: "Europe",
          population: 621718
        }
        
        const { container } = render(<CountryCard info={info}/>);
        // eslint-disable-next-line testing-library/no-node-access
        expect(container.firstChild).toMatchSnapshot();
    });
});