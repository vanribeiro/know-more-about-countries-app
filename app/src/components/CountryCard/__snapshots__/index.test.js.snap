// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should always have the same snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-iCfMLu eZlenB css-66l59o-MuiPaper-root-MuiCard-root"
>
  <img
    alt="Montenegro Flag"
    class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img sc-furwcr fksWRi css-o69gx8-MuiCardMedia-root"
    src="https://flagcdn.com/me.svg"
  />
  <div
    class="MuiCardContent-root sc-jrQzAO cCcBeS css-46bh2p-MuiCardContent-root"
  >
    <h2
      class="sc-pVTFL lixlSB"
    >
      Montenegro
    </h2>
    <p
      class="sc-hKwDye keOyrB"
    >
      <strong
        class="sc-dkPtRN erdiHc"
      >
        Population: 
      </strong>
      621,718
    </p>
    <p
      class="sc-hKwDye keOyrB"
    >
      <strong
        class="sc-dkPtRN erdiHc"
      >
        Region: 
      </strong>
      Europe
    </p>
    <p
      class="sc-hKwDye keOyrB"
    >
      <strong
        class="sc-dkPtRN erdiHc"
      >
        Capital: 
      </strong>
      Podgorica
    </p>
  </div>
</div>
`;
